src/main/java/com/fasterxml/jackson/databind/deser/std/CollectionDeserializer.java

```diff
@@ -424,4 +440,40 @@ public void handleResolvedForwardReference(Object id, Object value) throws IOExc
             _parent.resolveForwardReference(id, value);
         }
     }
+
+    @Override
+    public Collection<Object> getNullValue(DeserializationContext ctxt) throws JsonMappingException {
+        if (_readNullAsEmpty == Boolean.TRUE ||
+                ctxt.hasDeserializationFeatures(DeserializationFeature.READ_NULL_OR_MISSING_CONTAINER_AS_EMPTY.getMask())) {
```

Comment:
I think a more limited patch would only use the second part of the checks, as that would allow mapping explicit `null`s to empty Collection; and later on could probably be used to help with "map missing to empty" (although question there would be that of how to avoid features check...)